src/Disciplina/Core/Types.hs

```diff
@@ -62,15 +68,65 @@ newtype CourseId = CourseId
     { getCourseId :: Word32
     } deriving (Eq, Ord, Show, Num)
 
+-- | Assignment can be either regular of final
+data AssignmentType = Regular | CourseFinal
+    deriving (Eq, Show, Generic)
+
+-- | Assignment doesn't contain actual assignment contents - only hash of them.
+data Assignment = Assignment
+    { aCourseId   :: !CourseId
+    -- ^ Course this assignement belongs to
+    , aType       :: !AssignmentType
+    -- ^ Assignment type
+    , aAssignment :: !Text
+    -- ^ Hash of assignment content
```

Comment:
No, this `Text` field is actually supposed to be a short description of an assignment, and a _hash_ of an assignment is a separate field (so the assignment ID is a hash of its contents, not a hash of this datatype).
However, I'm thinking about maybe leaving assignment ID a hash of a datatype, and providing a hash of assignment contents separately - this way more data will be validated by student's submission signature. I'll think about it and tell my decision a little bit later.